universal-application-tool-0.0.1/conf/routes

```diff
@@ -8,11 +8,13 @@ GET     /playIndex                  controllers.HomeController.playIndex()
 GET     /securePlayIndex            controllers.HomeController.securePlayIndex()
 
 # A controller for pages for an admin to create and maintain programs
-GET     /admin/programs                    controllers.admin.AdminProgramController.index
-GET     /admin/programs/new                controllers.admin.AdminProgramController.newOne(request: Request)
-GET     /admin/programs/:programId/edit    controllers.admin.AdminProgramController.edit(request: Request, programId: Long)
-POST    /admin/programs                    controllers.admin.AdminProgramController.create(request: Request)
-POST    /admin/programs/:programId         controllers.admin.AdminProgramController.update(request: Request, programId: Long)
+GET     /admin/programs                       controllers.admin.AdminProgramController.index
+GET     /admin/programs/new                   controllers.admin.AdminProgramController.newOne(request: Request)
+GET     /admin/programs/:programId/edit       controllers.admin.AdminProgramController.edit(request: Request, programId: Long)
+GET     /admin/programs/:programId/publish    controllers.admin.AdminProgramController.publish(request: Request, programId: Long)
+GET     /admin/programs/:programId/newVersion controllers.admin.AdminProgramController.newVersionFrom(request: Request, programId: Long)
```

Comment:
Suggestion (blocking): GET requests should not generally create or update things. Recommend changing these to POST and using submit buttons in the view.

See Aaron's use of submit buttons with hidden fields as an example:
[button](https://github.com/seattle-uat/civiform/blob/main/universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java#L132-L136)
[hidden form](https://github.com/seattle-uat/civiform/blob/main/universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockEditView.java#L69-L70)